modules/roles.js

```diff
@@ -0,0 +1,202 @@
+*/
+async function roleFunc(int, give = true) {
+  try {
+    await int.deferReply({ ephemeral: true });
+    const input = int.options.getString("role", true);
+    if (u.perms.isAdmin(int.member)) {
+      const member = int.guild.members.cache.find(usr => usr.id === int.user.id);
+      const role = int.guild.roles.cache.find(r => r.name.toLowerCase() == input.toLowerCase());
+      if (role) {
+        if (give) {
+          try {
+            if (member?.roles.cache.has(role.id)) return int.editReply(`You already have the ${role} role`);
+            await member?.roles.add(role);
+            return int.editReply(`Successfully added the ${role} role`);
+          } catch (e) { return int.editReply(`Failed to add the ${role} role`); }
+        }
+        try {
+          if (!member?.roles.cache.has(role.id)) return int.editReply(`You do not have the ${role} role`);
+          await member?.roles.remove(role);
+          return int.editReply(`Successfully removed the ${role} role`);
+        } catch (e) { return int.editReply(`Failed to remove the ${role} role`); }
+      } else { return int.editReply(`I couldn't find the ${input} role`); }
+    } else if (roles.has(input.toLowerCase())) {
+      const role = int.guild.roles.cache.get(roles.get(input.toLowerCase()));
+      const member = await int.guild.members.fetch(int.user.id);
+      if (role) {
+        if (give) {
+          try {
+            if (member) await member.roles.add(role);
+            return int.editReply(`Successfully added the ${role} role`);
+          } catch (e) { return int.editReply(`Failed to add the ${role} role`); }
+        }
+        try {
+          if (member) await member.roles.remove(role);
+          return int.editReply(`Successfully removed the ${role} role`);
+        } catch (e) { return int.reply(`Failed to remove the ${role} role`); }
+      }
+    } else {
+      return int.editReply(`You didn't give me a valid role`);
+    }
+  } catch (error) { u.errorHandler(error, int); }
+}
+
+/** @param {Augur.GuildInteraction<"CommandSlash">} int */
+async function whoHas(int) {
+  try {
+    const input = int.options.getString("role", true);
+    const role = int.guild.roles.cache.find(r => r.name.toLowerCase() == input.toLowerCase());
+    if (role) {
+      if (role && role.members.size > 0) int.reply({ content: `Members with the ${role} role:\n\`\`\`\n${role.members.map(m => m.displayName).sort().join("\n")}\n\`\`\``, ephemeral: !(int.channel?.id == int.client.getTextChannel(u.sf.channels.general)?.id) });
+      else int.reply({ content: "I couldn't find any members with that role. :shrug:", ephemeral: true });
+    } else {
+      int.reply({ content: "You need to give me a valid role to find!", ephemeral: true });
+      u.clean(int);
+    }
+  } catch (error) { u.errorHandler(error, int); }
+}
+
+/**
+ * @param {Augur.GuildInteraction<"CommandSlash">} int
+ * @param {Boolean} give
+*/
+async function staffFunc(int, give = true) {
```

Comment:
could you move most of this function to modCommon? we might put this in mod menu at some point. have the function return a reply option object or string and handle interaction responses in this file. look at pretty much anything in 01-mods.js for a good example